src/Text/Pandoc/Writers/LaTeX.hs

```diff
@@ -65,10 +65,37 @@ import Text.Pandoc.Writers.LaTeX.Util (stringToLaTeX, StringContext(..),
 import Text.Pandoc.Writers.Shared
 import qualified Text.Pandoc.Writers.AnnotatedTable as Ann
 
+isolateBigNotes :: ([Inline] -> Inline) -> [Inline] -> [Inline]
+isolateBigNotes constructor xs =
+  let (before, after) = break isBigNote xs
+  in case after of
+       (noteInline:rest) -> constructor before :
+                            noteInline :
+                            isolateBigNotes constructor rest
+       [] -> [constructor xs]
+  where
+    isBigNote :: Inline -> Bool
+    isBigNote (Note [Plain _]) = False  -- A small note
+    isBigNote (Note [Para _]) =  False  -- A small note
+    isBigNote (Note _) = True  -- A big note
+    isBigNote _ = False  -- Not a note
+
+raiseBigNotes :: [Inline] -> [Inline]
+raiseBigNotes (Emph inner : xs)
+  = isolateBigNotes Emph (raiseBigNotes inner) ++ raiseBigNotes xs
+raiseBigNotes (Strong inner : xs)
+  = isolateBigNotes Strong (raiseBigNotes inner) ++ raiseBigNotes xs
+raiseBigNotes (Underline inner : xs)
+  = isolateBigNotes Underline (raiseBigNotes inner) ++ raiseBigNotes xs
+raiseBigNotes (Strikeout inner : xs)
+  = isolateBigNotes Strikeout (raiseBigNotes inner) ++ raiseBigNotes xs
+raiseBigNotes (x : xs)          = x : raiseBigNotes xs
+raiseBigNotes [] = []
+
```

Comment:
I'm getting a bit worried about the performance implications of this.
Can you run `make bench` with and without your PR to see what change we see in the LaTeX writer results?

Reply:
Below are the results of the benchmark tests. It looks like there is no performance regression. What do you think?
- Before my change
    - 1 st trial (I don’t know why it was slow.)
        
        ```jsx
        latex:                 OK (4.77s)
        5.10 ms ± 4.5 ms,  13 MB allocated, 435 KB copied, 122 MB peak memory,       same as baseline
        ```
        
    - 2nd trial
        
        ```jsx
        latex:                 OK (2.48s)
        2.49 ms ± 318 μs,  13 MB allocated, 442 KB copied, 128 MB peak memory,       same as baseline
        ```
        
- After my change
    - 1st trial
        
        ```jsx
        latex:                 OK (2.77s)
        2.80 ms ± 253 μs,  14 MB allocated, 538 KB copied, 124 MB peak memory,       same as baseline
        ```
        
    - 2nd trial
        
        ```jsx
        latex:                 OK (2.72s)
        2.79 ms ± 136 μs,  14 MB allocated, 541 KB copied, 124 MB peak memory,       same as baseline
        ```

Reply:
Assuming we throw out the first result as anomalous, it looks like there's a slight performance regression (on order of 10%).

If that's right, then it's a question of weighing a slight performance hit on all uses vs better correctness for a rare case (a note inside an emphasized section).